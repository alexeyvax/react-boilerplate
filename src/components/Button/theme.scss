@import '~common';

button {
  @include reset-button();

  padding: 10px 14px;
  background-color: $button-color;
}

.testing {
  display: block;
}

.clear {
  display: block;
}
