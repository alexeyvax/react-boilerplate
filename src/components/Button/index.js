import React from 'react';
import propTypes from 'prop-types';

import theme from './theme.scss';

const Button = ({ content, className, handler }) => (
  <button
    className={theme[className]}
    type="button"
    onClick={handler}
  >
    {content}
  </button>
);

Button.propTypes = {
  content: propTypes.string,
  className: propTypes.string,
  handler: propTypes.func.isRequired,
};

Button.defaultProps = {
  content: 'button',
  className: null,
};

export default Button;
