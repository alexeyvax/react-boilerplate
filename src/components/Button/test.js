import React from 'react';
import renderer from 'react-test-renderer';

import Button from '.';

it('renders correctly without optional props', () => {
  const tree = renderer
    .create(<Button handler={jest.fn()} />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders correctly with optional props', () => {
  const tree = renderer
    .create(<Button content="Button name" className="test" handler={jest.fn()} />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
