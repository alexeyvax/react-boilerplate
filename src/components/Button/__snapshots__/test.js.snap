// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with optional props 1`] = `
<button
  className="test"
  onClick={[MockFunction]}
  type="button"
>
  Button name
</button>
`;

exports[`renders correctly without optional props 1`] = `
<button
  className="null"
  onClick={[MockFunction]}
  type="button"
>
  button
</button>
`;
